ykman-gui/qml/PivSetManagementKeyForm.qml

```diff
@@ -0,0 +1,247 @@
+        id: tabs
+        Layout.fillWidth: true
+
+        Layout.minimumHeight: Utils.sum(Utils.pick(contentItem.children,
+                                                   'implicitHeight'))
+        Layout.minimumWidth: Utils.sum(Utils.pick(contentItem.children,
+                                                  'implicitWidth'))
+
+        Component.onCompleted: {
+            tabs.currentIndex = 1
+            tabs.currentIndex = 0
+        }
+
+        Tab {
+            id: personalTab
+            title: qsTr("Personal")
+            anchors.margins: margins
+
+            ColumnLayout {
+                Label {
+                    text: qsTr("PIN will be used as management key.")
+                }
+            }
+        }
+
+        Tab {
+            id: enterpriseTab
+            title: qsTr("Enterprise")
+            anchors.margins: margins
+
+            ColumnLayout {
+                id: foo
+
+                Label {
+                    text: qsTr("A management key separate from the PIN will be created.")
+                }
+
+                Label {
+                    text: qsTr("New management key:")
+                }
+
+                TextField {
+                    id: newManagementKeyInput
+                    maximumLength: 48
+                    focus: true
+                    Layout.fillWidth: true
+
+                    onTextChanged: setNewManagementKeyInput(
+                                       text) // Root can't reference TabView children directly
+
+                    validator: RegExpValidator {
+                        regExp: /[0-9a-f]{48}/
+                    }
+                }
+
+                RowLayout {
+                    Button {
+                        text: qsTr("Randomize")
+                        onClicked: {
+                            device.piv_generate_random_mgm_key(function (key) {
+                                newManagementKeyInput.text = key
+                            })
+                        }
+                    }
+
+                    Button {
+                        text: qsTr("Copy to clipboard")
+                        onClicked: copyToClipboard(newManagementKeyInput.text)
+                    }
+                }
+            }
+        }
+    }
+
+    RowLayout {
+
+        Label {
+            Layout.fillWidth: true
+            enabled: !pinIsKeyCurrently
+            text: qsTr("Current management key:")
+        }
+
+        Label {
+            Layout.alignment: Qt.AlignRight
+            font.italic: true
+            text: qsTr("PIN is currently used as key.")
+            visible: pinIsKeyCurrently
+        }
+    }
+
+    RowLayout {
+        TextField {
+            id: currentManagementKeyInput
+            enabled: !pinIsKeyCurrently
+            maximumLength: 48
+            Layout.fillWidth: true
+
+            validator: RegExpValidator {
+                regExp: /[0-9a-f]{48}/
+            }
+        }
+
+        Button {
```

Comment:
Maybe we could recognise that the default management key is in use instead.

Reply:
Yeah, good idea. Shall we do that in a separate PR?